### PR TITLE
Adding Legendre-to-Chebyshev C/python package

### PR DESCRIPTION
**Pull request by mikaem**
Checklist
- [x] Title of this PR is meaningful: e.g. "Adding my_nifty_package", not "updated meta.yaml".
- [x] License file is packaged (see [here](https://github.com/conda-forge/staged-recipes/blob/5eddbd7fc9d1502169089da06c3688d9759be978/recipes/example/meta.yaml#L64-L73) for an example).
- [x] Source is from official source.
- [x] Package does not vendor other packages. (If a package uses the source of another package, they should be separate packages or the licenses of all packages need to be packaged).
- [x] If static libraries are linked in, the license of the static library is packaged.
- [x] Package does not ship static libraries. If static libraries are needed, [follow CFEP-18](https://github.com/conda-forge/cfep/blob/main/cfep-18.md).
- [x] Build number is 0.
- [x] A tarball (`url`) rather than a repo (e.g. `git_url`) is used in your recipe (see [here](https://conda-forge.org/docs/maintainer/adding_pkgs.html#build-from-tarballs-not-repos) for more details).
- [x] GitHub users listed in the maintainer section have posted a comment confirming they are willing to be listed there.
- [x] When in trouble, please check our [knowledge base documentation](https://conda-forge.org/docs/maintainer/knowledge_base.html) before pinging a team.

### CODE REVIEW

**Comment by mikaem** [comment]
@conda-forge/help-python-c, ready for review!

File: recipes/leg2cheb/meta.yaml [hunk 1]
```diff
@@ -0,0 +1,62 @@
...
+package:
+  name: {{ name }}
+  version: {{ version }}
+
+source:
+  url: https://github.com/spectralDNS/Legendre-to-Chebyshev/archive/{{ version }}.tar.gz
+  sha256: 3ff70cecdb98cb851dc8b04709c616cdd6b3b681dee83c0c8b022a6422718189
+
+build:
+  number: {{ build }}
```
Comment (xhochy) [inline recipes/leg2cheb/meta.yaml:14]:
```suggestion
  number: 0
```

File: recipes/leg2cheb/meta.yaml [hunk 2]
```diff
@@ -0,0 +1,62 @@
+{% set build = 0 %}
```
Comment (xhochy) [inline recipes/leg2cheb/meta.yaml:1]: No need to make this Jinja
```suggestion
```

File: recipes/leg2cheb/meta.yaml [hunk 3]
```diff
@@ -0,0 +1,62 @@
...
+    - meson compile -C build
+    - meson install -C build
+
+requirements:
+  build:
+    - numpy >=2.0                              
+    - setuptools                          
+    - python                              
+    - cython                              
+    - pip
```
Comment (xhochy) [inline recipes/leg2cheb/meta.yaml:28]: These are all host dependencies
```suggestion
```

File: recipes/leg2cheb/meta.yaml [hunk 4]
```diff
@@ -0,0 +1,62 @@
...
+    - numpy >=2.0                              
+    - setuptools                          
+    - python                              
+    - cython                              
+    - pip                                 
+    - meson                               
+    - ninja                               
+    - pkg-config         
+    - llvm-openmp  # [osx]
+    - libgomp      # [linux]
```
Comment (xhochy) [inline recipes/leg2cheb/meta.yaml:33]: Please add these two additionally to host
```suggestion
    - llvm-openmp  # [osx]
    - libgomp      # [linux]            
```

File: recipes/leg2cheb/meta.yaml [hunk 5]
```diff
@@ -0,0 +1,62 @@
...
+    - setuptools                          
+    - python                              
+    - cython                              
+    - pip                                 
+    - meson                               
+    - ninja                               
+    - pkg-config         
+    - llvm-openmp  # [osx]
+    - libgomp      # [linux]            
+    - compilers
```
Comment (xhochy) [inline recipes/leg2cheb/meta.yaml:34]: compilers is a package only for endusers, please use the compiler functions in recipes.
```suggestion
    - {{ compiler('c') }}
```

File: recipes/leg2cheb/meta.yaml [hunk 6]
```diff
@@ -0,0 +1,62 @@
...
+
+  host:
+    - python
+    - cython
+    - fftw
+    - openblas
+    - libopenblas =*=*openmp*
+  run:
+    - python
+    - numpy >=2.0
```
Comment (xhochy) [inline recipes/leg2cheb/meta.yaml:45]: If numpy is in host, it doesn't not need to be listed in run as it has a run_exports.
```suggestion
```

File: recipes/leg2cheb/meta.yaml [hunk 7]
```diff
@@ -0,0 +1,62 @@
...
+    - fftw
+    - openblas
+    - libopenblas =*=*openmp*
+
+test:
```
Comment (xhochy) [inline recipes/leg2cheb/meta.yaml:50]:
```suggestion
test:
  requires:
    - pip
  commands:
    - pip check
```

**Review by xhochy — CHANGES_REQUESTED** [review]

File: recipes/leg2cheb/meta.yaml (hunk 4 above)
Comment (mikaem) [reply recipes/leg2cheb/meta.yaml:27]: Adding in [commit](https://github.com/conda-forge/staged-recipes/pull/27543/commits/5c4ad3b9ac574d4fea4d630313ce89e3d1aa6ef3)

File: recipes/leg2cheb/meta.yaml [hunk 8]
```diff
@@ -0,0 +1,64 @@
...
+  skip: true  # [win]
+  script: 
+    - cd src
+    - meson setup build ${MESON_ARGS}
+    - meson compile -C build
+    - meson install -C build
+
+requirements:
+  build:
+    - numpy >=2.0
```
Comment (xhochy) [inline recipes/leg2cheb/meta.yaml:23]: This needs to be in host
```suggestion
```

File: recipes/leg2cheb/meta.yaml [hunk 9]
```diff
@@ -0,0 +1,64 @@
...
+  host:
+    - python
+    - cython
+    - fftw
+    - openblas
+    - libopenblas =*=*openmp*
+    - llvm-openmp  # [osx]
+    - libgomp      # [linux]
+  run:
+    - numpy >=2.0
```
Comment (xhochy) [inline recipes/leg2cheb/meta.yaml:42]: Numpy has a run_exports and will add the correct run_dependency if it is in host
```suggestion
```

**Review by xhochy — CHANGES_REQUESTED** [review]

File: recipes/leg2cheb/meta.yaml [hunk 10]
```diff
@@ -0,0 +1,61 @@
...
+source:
+  url: https://github.com/spectralDNS/Legendre-to-Chebyshev/archive/{{ version }}.tar.gz
+  sha256: 3ff70cecdb98cb851dc8b04709c616cdd6b3b681dee83c0c8b022a6422718189
+
+build:
+  number: 0
+  skip: true  # [win]
+  script: 
+    - cd src
+    - meson setup build ${MESON_ARGS}
```
Comment (xhochy) [inline recipes/leg2cheb/meta.yaml:17]: This should show you better what the error is. I can highly recommend moving this to a `build.sh` file, though.
```suggestion
    - meson setup build ${MESON_ARGS} || (cat $SRC_DIR/src/build/meson-logs/meson-log.txt; exit 1)
```

Comment (mikaem) [reply recipes/leg2cheb/meta.yaml:17]: Thanks @xhochy, I'll try that.

**Comment by mikaem** [comment]
I'm getting in the [linux build](https://dev.azure.com/conda-forge/feedstock-builds/_build/results?buildId=1032757&view=logs&j=6f142865-96c3-535c-b7ea-873d86b887bd&t=22b0682d-ab9e-55d7-9c79-49f3c3ba4823&l=1152):

    ModuleNotFoundError: No module named 'Cython'

and in the [osx build](https://dev.azure.com/conda-forge/feedstock-builds/_build/results?buildId=1032757&view=logs&j=e35d4f76-8ff2-5536-d795-df91e63eb9f7&t=fa7b4b17-b6ff-5c9c-8cfc-15f888c92310&l=2052):

    ModuleNotFoundError: No module named 'numpy'

Not sure what happens. Am I using the wrong meson or something?

**Comment by mikaem** [comment]
@xhochy Moving, meson, ninja and pkg-config from build to host fixed the problems and the build is now fine.

**Comment by xhochy** [comment]
Only meson should be in host, `ninja` and `pkg-config` need to be in build for cross-compilation to work.

**Comment by mikaem** [comment]
@xhochy should be ok now? Moved ninja and pkg-config to build and the checks are green.

**Review by xhochy — APPROVED** [review]

**Pull request merged**